pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+                        operation1.get_matrix(),
+                    ),
+                    np.matmul(
+                        operation1.get_matrix(),
+                        qml.Rot(*operation2.data, wires=target_wires_2).get_matrix(),
+                    ),
+                )
+            )
+
+        return np.all(
+            np.allclose(
+                np.matmul(
+                    operation1.get_matrix(),
+                    operation2.get_matrix(),
+                ),
+                np.matmul(
+                    operation2.get_matrix(),
+                    operation1.get_matrix(),
+                ),
+            )
+        )
+    return None
+
+
+not_supported_operations = [
+    "PauliRot",
+    "QubitDensityMatrix",
+    "CVNeuralNetLayers",
+    "ApproxTimeEvolution",
+    "ArbitraryUnitary" "CommutingEvolution",
+    "DisplacementEmbedding",
+    "SqueezingEmbedding",
+]
+non_commuting_operations = [
+    "ArbitraryStatePreparation",
+    "BasisStatePreparation",
+    "MottonenStatePreparation",
+    "QubitCarry",
+    "QubitSum",
+    "SingleExcitation",
+    "SingleExcitationMinus",
+    "SingleExcitationPlus",
+    "DoubleExcitation",
+    "DoubleExcitationPlus",
+    "DoubleExcitationMinus",
+    "BasicEntanglerLayers",
+    "GateFabric",
+    "ParticleConservingU1",
+    "ParticleConservingU2",
+    "RandomLayers",
+    "SimplifiedTwoDesign",
+    "StronglyEntanglingLayers",
+    "AllSinglesDoubles",
+    "FermionicDoubleExcitation",
+    "FermionicSingleExcitation",
+    "Grover",
+    "kUpCCGSD",
+    "Permute",
+    "QFT",
+    "QuantumMonteCarlo",
+    "QuantumPhaseEstimation",
+    "UCCSD",
+    "MPS",
+    "TTN",
+    "AmplitudeEmbedding",
+    "AngleEmbedding",
+    "BasisEmbedding",
+    "IQPEmbedding",
+    "QAOAEmbedding",
+]
+
+
+def is_commuting(operation1, operation2):
+    r"""Check if two operations are commuting. A lookup table is used to check the commutation between the
+    controlled, targetted part of operation 1 with the controlled, targetted part of operation 2. It supports
+    most PennyLane operations that are not CV operations.
+
+    Args:
+        operation1 (.Operation): A first quantum operation.
+        operation2 (.Operation): A second quantum operation.
+
+    Returns:
+         bool: True if the operations commute, False otherwise.
+
+    **Example**
+
+    >>> qml.is_commuting(qml.PauliX(wires=0), qml.PauliZ(wires=0))
+    False
+    """
+    # pylint: disable=too-many-branches
+    # pylint: disable=too-many-return-statements
+
+    control_base = {
+        "CNOT": "PauliX",
+        "CZ": "PauliZ",
+        "CY": "PauliY",
+        "CSWAP": "SWAP",
+        "Toffoli": "PauliX",
+        "ControlledPhaseShift": "PhaseShift",
+        "CRX": "RX",
+        "CRY": "RY",
+        "CRZ": "RZ",
+        "CRot": "Rot",
+        "MultiControlledX": "PauliX",
+        "ControlledOperation": "ControlledOperation",
+    }
```

Comment:
Should this be defined at the module level?

Reply:
I don't think it is necessary and it will be removed as soon as the control refractor is done!